wandb/apis/public.py

```diff
@@ -1916,22 +1916,27 @@ def file(self, name):
     def upload_file(self, path, root="."):
         """
         Arguments:
-            path (str): name of file to upload.
-            root (str): the root path to save the file relative to.  i.e.
-                If you want to have the file saved in the run as "my_dir/file.txt"
-                and you're currently in "my_dir" you would set root to "../"
+            path (str): local path to the file to upload.
+            root (str): the root path to save the file in wandb.
 
         Returns:
             A `File` matching the name argument.
         """
+        path = os.path.normpath(os.path.join(os.getcwd(), os.path.relpath(path, os.getcwd())))
+        if not os.path.isfile(path):
+            raise Exception(f"File {path} doesn't exist!")
+        if root == "/":
+            raise Exception("'/' is not a valid directory, to use the root directory use '.' instead.")
+        root_ext = os.path.splitext(root)[1]
+        if root_ext not in [os.path.splitext(path)[1], ""]:
+            raise Exception(f"Local file extension doesn't match target file extension!")
```

Comment:
Oops forgot to add the extensions to this f-string, I'll do it later today!

Reply:
```suggestion
            raise Exception(f"Local file extension ({root_ext}) doesn't match target file extension ({target_ext})!")
```

Reply:
Suggestion above won't work yet as ye first need to define `target_ext` value as `os.path.splitext(path)[1]`

Reply:
Indeed we've created `path_ext` instead of `target_ext` as the name was misleading since the target file is `root`, `path` is the one that exists locally already.